webapp/src/services/api.ts

```diff
@@ -224,15 +216,15 @@ export const api = createApi({
         "UtteranceCountPerFilter",
         "Utterances",
       ],
-      async onQueryStarted(
-        { persistentIds, newValue, ...args },
-        { dispatch, queryFulfilled }
-      ) {
+      async onQueryStarted({ body, ...args }, { dispatch, queryFulfilled }) {
         const patchResult = dispatch(
           api.util.updateQueryData("getUtterances", args, (draft) => {
             draft.utterances.forEach((utterance) => {
-              if (persistentIds.includes(utterance.persistentId)) {
-                utterance.dataAction = newValue;
+              const patchUtterance = body.find(
+                ({ persistentId }) => persistentId === utterance.persistentId
```

Comment:
Follow up on [this](https://github.com/ServiceNow/azimuth/pull/465#discussion_r1126572158), [this](https://github.com/ServiceNow/azimuth/pull/465#discussion_r1126975146) and [this](https://github.com/ServiceNow/azimuth/pull/465#discussion_r1128000847).
There is still an issue. Have you tried it with a CSV file to make sure it works?
The `utterance.persistentId` can be a `string` or a `number`. However, when reading the CSV file, there is no distinction between a `string` and a `number`, so we assume `persistentId` is a string. The back end supports that, no problem. Here, we just have to check equality with `==` instead of `===` to support that.
```suggestion
                // Support number | string
                // eslint-disable-next-line eqeqeq
                ({ persistentId }) => persistentId == utterance.persistentId
```

Reply:
Yes, it did work for me with the sample import of data. It's good to use equality to avoid the type-checking, but I wonder why the type would be changed in the CSV

Reply:
There is no typing in CSV. There is no way to tell if `...,42,...` is the number `42` or the string `"42"`. Does that make sense?

Reply:
Yes :)

Reply:
Please add the comments I suggested, otherwise we get a warning in the terminal.